fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -0,0 +1,222 @@
+                    throw new SemanticException("Materialized view partition function " +
+                            functionName + " must match pattern:" + mvColumnPattern);
+                }
+            }
+        }
+
+        private void checkPartitionKey(PartitionExpDesc partitionExpDesc, Map<String, TableRelation> tableRelationHashMap) {
+            SlotRef slotRef = null;
+            Expr expr = partitionExpDesc.getExpr();
+            if (expr instanceof FunctionCallExpr) {
+                slotRef = (SlotRef) expr.getChild(0);
+            } else {
+                slotRef = ((SlotRef) expr);
+            }
+            //must have table relation
+            TableRelation tableRelation = tableRelationHashMap.get(slotRef.getTblNameWithoutAnalyzed().getTbl());
+            PartitionInfo partitionInfo = ((OlapTable) tableRelation.getTable()).getPartitionInfo();
+            if (partitionInfo instanceof SinglePartitionInfo) {
+                throw new SemanticException("Materialized view base table must have partition key");
+            }
+            RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+            List<Column> partitionColumns = rangePartitionInfo.getPartitionColumns();
+            SlotRef finalSlotRef = slotRef;
+            if (partitionColumns.stream().filter(column -> {
+                return column.getName().equals(finalSlotRef.getColumnName());
+            }).count() == 0) {
+                throw new SemanticException("Materialized view partition key in partition exp " +
+                        "must be base table partition key");
+            }
+        }
+
+        private void checkDistribution(CreateMaterializedViewStatement statement) {
+            DistributionDesc distributionDesc = statement.getDistributionDesc();
+            Map<String, String> properties = statement.getProperties();
+            List<Column> mvColumnItems = statement.getMvColumnItems();
+            if (distributionDesc == null) {
+                if (ConnectContext.get().getSessionVariable().isAllowDefaultPartition()) {
+                    if (properties == null) {
+                        properties = Maps.newHashMap();
+                        properties.put(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM, "1");
+                    }
+                    distributionDesc = new HashDistributionDesc(Config.default_bucket_num,
+                            Lists.newArrayList(mvColumnItems.get(0).getName()));
+                    statement.setDistributionDesc(distributionDesc);
+                } else {
+                    throw new SemanticException("Materialized view should contain distribution desc");
+                }
+            }
+            try {
+                distributionDesc.analyze(mvColumnItems.stream().map(column -> column.getName()).collect(Collectors.toSet()));
+            } catch (AnalysisException e) {
+                throw new SemanticException(e.getMessage());
```

Comment:
you change the exception type here, better to add LOG.warning(e) here to print the old exception. or we will lose the stack.